src/test/java/org/takes/misc/VerboseIterableTest.java

```diff
@@ -0,0 +1,32 @@
+package org.takes.misc;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.collection.IsIterableWithSize;
+import org.junit.Test;
+
+/**
+ * 
+ * @author marcus.sanchez (sanchez.marcus@gmail.com)
+ * @version $Id$
+ * @since 0.15.1
+ */
+public class VerboseIterableTest {
+
+	private static final String NO_NEXT_VALUE_ERROR = "Empty Error Message";
+
+	private final List<String> VALID_LIST = Arrays.asList("Accept: text/plain",
+			"Accept-Charset: utf-8",
+			"Authorization: Basic QWxhZGRpbjpvcGVuIHNlc2FtZQ==",
+			"Cache-Control: no-cache", "From: user@example.com");
+
```

Comment:
@m4solutions Add a javadoc comment here with the form `VerboseIterable can ...`. I suggest:
```java
/**
 * VerboseIterable can return correct size.
 */
```